theopensystemslab/planx-new
# feat: generate CSV data on the backend so that we can send to Uniform using only a sessionId

depends on https://github.com/theopensystemslab/planx-core/pull/30 & builds on #1671

Changes:
- [x] adds new endpoint GET `/admin/session/:sessionId/csv` (use with optional param `?download=true`)
- [x] updates POST `/uniform/:localAuthority` to only require a sessionId to be passed through from createSendEvents, no longer the full `csv` data string

To test:
- confirm you can submit an LDC app to Uniform staging successfully!
  - flow: https://1695.planx.pizza/buckinghamshire/apply-for-a-lawful-development-certificate/preview
  - event should only have sessionId in payload and succeed: https://hasura.1695.planx.pizza/console/events/one-off-scheduled-events/processed
   - audit table should have a success record: https://hasura.1695.planx.pizza/console/data/default/schema/public/tables/uniform_applications/browse
  
Future cleanup/considerations:
- Replace all instances of `makeCsvData` & `getBOPSParams` methods in Editor using `$public` client methods - will require headers to access session data first! 
- When we make the switch to `$public.generateCSVData`, we'll only be able to generate CSVs for sessions using `lowcal_sessions`, not local storage. Do we have any existing instances of offering CSVs for FOIYNPP or elsewhere for non-S&R services? I don't think there's any Confirmation components used in those services?
